### DISCUSSION TITLE
Refetch with enabled: false executes request but data is undefined

### DISCUSSION THREAD

**Discussion by gdelacruzfdez** [TanStack/query#1273] [Q&A] [answered]
Hello,
I am having an issue in react-query 3.1.1. I have this query that I call manually using refetch:

```javascript
const { data, refetch, isSuccess } = useQuery(
    ['alarms', lowerLimitDate, upperLimitDate, selectedSources],
    () =>
      AlarmExtractionAPI.getAlarms({
        dateBegin: lowerLimitDate,
        dateEnd: upperLimitDate,
        sources: selectedSources
      }).then(res => res.data),
    { enabled: false }
  );
```
When I execute refetch the request is successfully made, but data is still undefined and isSuccess is false (only isIdle and isStale are set to true)

Is this a bug or I am doing something wrong?

Thank you!

**Comment by TkDodo** [comment, marked as answer]
Hmm, the docs explicitly mention that this should work:

https://github.com/tannerlinsley/react-query/blob/25a776d98dfbd629b301abce4dc9b76e5409c881/src/core/types.ts#L72-L77

So maybe it is a bug. Could you create a reproducible codesandbox that shows that behaviour?

As an alternative, instead of triggering `refetch` manually, you could keep a `state` boolean variable and pass that to `enabled`. Then, when you want to trigger the refetch, just set the boolean to true, which will trigger a re-render and the query will then be `enabled` and fetch normally. This is how I've been handling queries that should be disabled until some action is taken.

But I would still be interested if you can reproduce that bug in codesandbox :)

**Reply by TkDodo** [reply]
so I forked the simple example, disabled the query and made a button that re-fetches: https://codesandbox.io/s/zen-darkness-vfdjf
and here is the same example in v3 beta: https://codesandbox.io/s/romantic-torvalds-86bbh
it seems to work as advertised.

**Reply by gdelacruzfdez** [reply]
Yes, it works as expected. After investigating the problem was inside my fetch function and it was failing silently.

Thank you very much for your time!
